pagerduty/resource_pagerduty_event_orchestration_path_router.go

```diff
@@ -0,0 +1,393 @@
+}
+
+func expandRouterConditions(v interface{}) []*pagerduty.EventOrchestrationPathRuleCondition {
+	var conditions []*pagerduty.EventOrchestrationPathRuleCondition
+
+	for _, cond := range v.([]interface{}) {
+		c := cond.(map[string]interface{})
+
+		cx := &pagerduty.EventOrchestrationPathRuleCondition{
+			Expression: c["expression"].(string),
+		}
+
+		conditions = append(conditions, cx)
+	}
+
+	return conditions
+}
+
+func expandRouterActions(v interface{}) *pagerduty.EventOrchestrationPathRuleActions {
+	var actions = new(pagerduty.EventOrchestrationPathRuleActions)
+	for _, ai := range v.([]interface{}) {
+		am := ai.(map[string]interface{})
+		actions.RouteTo = am["route_to"].(string)
+	}
+
+	return actions
+}
+
+func expandCatchAll(v interface{}) *pagerduty.EventOrchestrationPathCatchAll {
+	var catchAll = new(pagerduty.EventOrchestrationPathCatchAll)
+
+	for _, ca := range v.([]interface{}) {
+		am := ca.(map[string]interface{})
+		catchAll.Actions = expandRouterActions(am["actions"].(interface{}))
+	}
+
+	return catchAll
+}
+
+func flattenSets(orchPathSets []*pagerduty.EventOrchestrationPathSet) []interface{} {
+	var flattenedSets []interface{}
+
+	for _, set := range orchPathSets {
+		flattenedSet := map[string]interface{}{
+			"id":    set.ID,
+			"rules": flattenRules(set.Rules),
+		}
+		flattenedSets = append(flattenedSets, flattenedSet)
+	}
+	return flattenedSets
+}
+
+func flattenRules(rules []*pagerduty.EventOrchestrationPathRule) []interface{} {
+	var flattenedRules []interface{}
+
+	for _, rule := range rules {
+		flattenedRule := map[string]interface{}{
+			"id":         rule.ID,
+			"label":      rule.Label,
+			"disabled":   rule.Disabled,
+			"conditions": flattenRouterConditions(rule.Conditions),
+			"actions":    flattenRouterActions(rule.Actions),
+		}
+		flattenedRules = append(flattenedRules, flattenedRule)
+	}
+
+	return flattenedRules
+}
+
+func flattenRouterConditions(conditions []*pagerduty.EventOrchestrationPathRuleCondition) []interface{} {
+	var flattendConditions []interface{}
+
+	for _, condition := range conditions {
+		flattendCondition := map[string]interface{}{
+			"expression": condition.Expression,
+		}
+		flattendConditions = append(flattendConditions, flattendCondition)
+	}
+
+	return flattendConditions
+}
+
+func flattenRouterActions(actions *pagerduty.EventOrchestrationPathRuleActions) []map[string]interface{} {
+	var actionsMap []map[string]interface{}
+
+	am := make(map[string]interface{})
+	am["route_to"] = actions.RouteTo
+	actionsMap = append(actionsMap, am)
+	return actionsMap
+}
+
+func flattenCatchAll(catchAll *pagerduty.EventOrchestrationPathCatchAll) []map[string]interface{} {
+	var caMap []map[string]interface{}
+
+	c := make(map[string]interface{})
+
+	c["actions"] = flattenRouterActions(catchAll.Actions)
+	caMap = append(caMap, c)
+
+	return caMap
+}
+
+func resourcePagerDutyEventOrchestrationPathRouterImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
```

Comment:
No tests for Import from what I searched. We might have to run an Import once all resources are ready to verify this.